sandpack-react/src/contexts/sandpackContext.tsx

```diff
@@ -389,6 +391,45 @@ class SandpackProvider extends React.PureComponent<
     });
   };
 
+  closeFile = (path: string): void => {
+    if (this.state.openPaths.length === 1) {
+      return;
+    }
+
+    this.setState(({ openPaths, activePath }) => {
+      const newPaths = openPaths.filter((openPath) => openPath !== path);
+
+      return {
+        activePath: path === activePath ? newPaths[0] : activePath,
+        openPaths: newPaths,
+        editorState: "dirty",
+      };
+    });
+  };
+
+  deleteFile = (path: string): void => {
```

Comment:
I didn't get where you're using this method, what it's the purpose of it?

Reply:
Not exactly using inside the `sandpack`. But more like to expose this functionality from outside. At the moment it's not possible to remove a file from the bundler.